app/products/agents/components/agent_post/index.tsx

```diff
@@ -0,0 +1,244 @@
+    );
+
+    useEffect(() => {
+        // Subscribe to streaming events for this specific post
+        const handleStreamingUpdate = (state: StreamingState) => {
+            setStreamingState({...state});
+        };
+
+        const handleStreamingEnded = () => {
+            // Clear streaming state immediately to force component to use persisted data
+            // This ensures POST_EDITED updates are reflected immediately
+            setStreamingState(undefined);
+        };
+
+        const startedListener = DeviceEventEmitter.addListener(
+            `${StreamingEvents.STARTED}_${post.id}`,
+            handleStreamingUpdate,
+        );
+
+        const updatedListener = DeviceEventEmitter.addListener(
+            `${StreamingEvents.UPDATED}_${post.id}`,
+            handleStreamingUpdate,
+        );
+
+        const endedListener = DeviceEventEmitter.addListener(
+            `${StreamingEvents.ENDED}_${post.id}`,
+            handleStreamingEnded,
+        );
+
+        return () => {
+            startedListener.remove();
+            updatedListener.remove();
+            endedListener.remove();
+        };
+    }, [post.id]);
+
+    // Determine the message to display
+    const displayMessage = streamingState?.message || post.message || '';
+    const isGenerating = streamingState?.generating ?? false;
+    const isPrecontent = streamingState?.precontent ?? false;
+
+    // Determine reasoning state - use streaming state if available, otherwise use persisted
+    const reasoningSummary = streamingState?.reasoning || persistedReasoning;
+    const isReasoningLoading = streamingState?.isReasoningLoading ?? false;
+    const showReasoning = streamingState?.showReasoning ?? (persistedReasoning !== '');
+
+    // Determine tool calls - use streaming state if available, otherwise use persisted
+    const toolCalls = streamingState?.toolCalls ?? persistedToolCalls;
+
+    // Determine annotations - use streaming state if available, otherwise use persisted
+    const annotations = streamingState?.annotations ?? persistedAnnotations;
+
+    // Check permissions
+    const isRequester = useMemo(() => {
+        return currentUserId ? isPostRequester(post, currentUserId) : false;
+    }, [post, currentUserId]);
+
+    // Determine if generation is in progress (generating or reasoning)
+    const isGenerationInProgress = isGenerating || isReasoningLoading;
+
+    // Show controls based on state and permissions
+    const showStopButton = isGenerationInProgress && isRequester;
+    const hasContent = displayMessage !== '' || reasoningSummary !== '';
+    const showRegenerateButton = !isGenerationInProgress && isRequester && hasContent;
+
+    // Handler for stop button
+    const handleStop = useCallback(async () => {
+        const {error} = await stopGeneration(serverUrl, post.id);
+        if (error) {
+            // Error is already logged in the action
+            // Could show a toast notification here if desired
```

Comment:
Are we still to make a decision on this?

Reply:
Fixed! Added toast notifications via showSnackBar for stop, regenerate, and tool approval errors